src/openrct2/object/RideObject.cpp

```diff
@@ -217,19 +217,15 @@ void RideObject::Load()
 
             if (vehicleEntry->car_visual != VEHICLE_VISUAL_RIVER_RAPIDS)
             {
-                int32_t b = vehicleEntry->base_num_frames * 32;
-
-                if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_USE_16_ROTATION_FRAMES)
-                    b /= 2;
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_USE_4_ROTATION_FRAMES)
-                    b /= 8;
+                const uint8_t numRotationFrames = vehicleEntry->GetNumRotationFrames();
+                int32_t b = vehicleEntry->base_num_frames * numRotationFrames;
```

Comment:
Been looking a bit at the code, might be better to rename ```b``` to something like ```rotatedImageCount```, the entire function could also be better structured probably, ```b``` b is also typically overwritten if one of the flags is set from its initial assignment, so moving this into a function on its own would help that.